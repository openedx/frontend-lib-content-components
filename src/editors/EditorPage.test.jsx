import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';
import EditorPage from './EditorPage';

const props = {
  courseId: 'course-v1:edX+DemoX+Demo_Course',
  blockType: 'html',
  blockId: 'block-v1:edX+DemoX+Demo_Course+type@html+block@030e35c4756a4ddc8d40b95fbbfff4d4',
  lmsEndpointUrl: 'evenfakerurl.com',
  studioEndpointUrl: 'fakeurl.com',
  onClose: jest.fn().mockName('props.onClose'),
};
jest.mock('react-redux', () => ({
  Provider: 'Provider',
  connect: (mapStateToProps, mapDispatchToProps) => (component) => ({
    mapStateToProps,
    mapDispatchToProps,
    component,
  }),
}));
jest.mock('./Editor', () => 'Editor');

describe('Editor Page', () => {
  describe('snapshots', () => {
    test('rendering correctly with expected Input', () => {
      expect(shallow(<EditorPage {...props} />).snapshot).toMatchSnapshot();
    });
    test('props besides blockType default to null', () => {
      expect(shallow(<EditorPage blockType={props.blockType} />).snapshot).toMatchSnapshot();
    });
  });
});
