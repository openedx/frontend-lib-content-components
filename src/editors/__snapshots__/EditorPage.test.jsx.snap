// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Page snapshots props besides blockType default to null 1`] = `
<Provider
  store={
    {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <ErrorBoundary
    learningContextId={null}
    studioEndpointUrl={null}
  >
    <Editor
      blockId={null}
      blockType="html"
      learningContextId={null}
      lmsEndpointUrl={null}
      onClose={null}
      returnFunction={null}
      studioEndpointUrl={null}
    />
  </ErrorBoundary>
</Provider>
`;

exports[`Editor Page snapshots rendering correctly with expected Input 1`] = `
<Provider
  store={
    {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <ErrorBoundary
    learningContextId="course-v1:edX+DemoX+Demo_Course"
    studioEndpointUrl="fakeurl.com"
  >
    <Editor
      blockId="block-v1:edX+DemoX+Demo_Course+type@html+block@030e35c4756a4ddc8d40b95fbbfff4d4"
      blockType="html"
      learningContextId="course-v1:edX+DemoX+Demo_Course"
      lmsEndpointUrl="evenfakerurl.com"
      onClose={[MockFunction props.onClose]}
      returnFunction={null}
      studioEndpointUrl="fakeurl.com"
    />
  </ErrorBoundary>
</Provider>
`;
